client/src/pages/RoomPage/index.tsx

```diff
@@ -157,7 +157,7 @@ function RoomPage({ name }: { name: string }): JSX.Element {
     });
 
     socket.on(SIOChannel.EDIT_USER_NAME, ({ id, name }) => {
-      editUserName(id, name);
+      addUserInfo(id)({ ...selfUserInfoRef.current, name })
```

Comment:
Sorry we weren't clear but this is super close.
You'll just need to do something like
```
socket.on(SIOChannel.EDIT_USER_NAME, ({ id, name }) => { 
    const newInfo = { ...selfUserInfoRef.current, name };
    selfUserInfoRef.current = newInfo;
    setSelfUserInfo(newInfo);
    addUserInfo(id)(newInfo);
});
```
to also update the `selfUserInfoRef.current`. Just tested it and it should work with this change👍

Reply:
Thanks for explaining and helping me thru this!

Reply:
Np thanks for baring with us haha!